#include <iostream>
using namespace std;

int divide(int dividend, int divisor) {
    if (divisor == 0) {
        throw "Divide by zero exception!";
    }
    return dividend / divisor;
}

int getElement(int arr[], int size, int index) {
    if (index < 0 || index >= size) {
        throw "Array index out of bounds exception!";
    }
    return arr[index];
}

int main() {
    try {
        int result = divide(10, 0);
        cout << "Result: " << result << endl;
    }
    catch (const char* exception) {
        cout << "Exception caught: " << exception << endl;
    }
    
    int numbers[] = { 1, 2, 3, 4, 5 };
    int size = sizeof(numbers) / sizeof(numbers[0]);
    
    try {
        int element = getElement(numbers, size, 10);
        cout << "Element: " << element << endl;
    }
    catch (const char* exception) {
        cout << "Exception caught: " << exception << endl;
    }
    
    return 0;
}
